pkg/apis/internal.acorn.io/v1/memory.go

```diff
@@ -0,0 +1,122 @@
+package v1
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"cuelang.org/go/cue/literal"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func ParseMemory(s []string) (Memory, error) {
+	result := Memory{}
+	for _, s := range s {
+		workload, memBytes, specific := strings.Cut(s, "=")
+
+		// If setting all, swap workload and memBytes
+		if !specific {
+			memBytes = workload
+			workload = ""
+		}
+
+		quantity, err := ParseQuantityString(memBytes)
+		if err != nil {
+			return Memory{}, err
+		}
+
+		result[workload] = &quantity
+	}
+	return result, nil
+}
+
+func ParseQuantityString(memory string) (int64, error) {
+	numInfo := literal.NumInfo{}
+	err := literal.ParseNum(memory, &numInfo)
+	if err != nil {
+		return -1, err
+	}
+
+	quantity, err := strconv.ParseInt(numInfo.String(), 10, 64)
+	if err != nil {
+		return -1, err
+	}
+
+	return quantity, nil
+}
+
+var (
+	ErrInvalidAcornMemory   = errors.New("invalid memory from acorn")
+	ErrInvalidSetMemory     = errors.New("invalid memory set by user")
+	ErrInvalidDefaultMemory = errors.New("invalid memory default")
+)
+
+type MemoryError struct {
+	Err      error
```

Comment:
https://github.com/acorn-io/acorn/pull/1028/files#annotation_8115132309 saying there is an invalid field type here

Reply:
Yeah I added a comment for deep copy gen to ignore this but I guess it didn't end up getting committed. Should be fixed now 👍🏻